summarization/web_entity_detection.py

```diff
@@ -47,34 +50,80 @@ def get_description_for_images(self, image_urls: list) -> list:
         list<str> : list containing best guess descriptions of the image
         '''
         self.image_urls = image_urls
+
+        # split into batches based on batch size
+        self._split_into_batches()
+
+        # make each request in a single thread
+        # this is done since request natively only
+        # allows one request per thread
+        self._make_concurrent_requests()
+
+        # we need to get the same order as
+        # that of the given image_urls
+        return self._get_ordered_and_combined_request_responses()
+
+    def _split_into_batches(self):
+        self.image_url_batches = list()
+        list_len = len(self.image_urls)
```

Comment:
change name for list_len to something like num_image_urls

Reply:
Will update accordingly.